usr/src/uts/armv8/os/cpuinfo_fdt.c

```diff
@@ -0,0 +1,445 @@
+/*
+ * This file and its contents are supplied under the terms of the
+ * Common Development and Distribution License ("CDDL"), version 1.0.
+ * You may only use this file in accordance with the terms of version
+ * 1.0 of the CDDL.
+ *
+ * A full copy of the text of the CDDL should have accompanied this
+ * source.  A copy of the CDDL is also available via the Internet at
+ * http://www.illumos.org/license/CDDL.
+ */
+
+/*
+ * Copyright 2023 Michael van der Westhuizen
+ */
+
+#include <sys/types.h>
+#include <sys/obpdefs.h>
+#include <sys/cpuinfo.h>
+#include <asm/controlregs.h>
+#include <sys/promif.h>
+#include <sys/cpuvar.h>
+#include <sys/byteorder.h>
+#include <sys/systm.h>
+#include <sys/stddef.h>
+
+#define	CPUINFO_STATUS_ERROR		-1
```

Comment:
Further down, you set these with `|=` as if they were bit specifications, but they aren't.

If they're not, can we assign them just with `=` and maybe use an enum?

Reply:
I'll disambiguate with an enum.

These values reflect the FDT node's `status` property (if present). The ORing is a set of derived flags that correspond to the flags in the ACPI 6.5 GICC CPU Interface Flags table (table 5.37, page 208). The (relevant at the moment) derived flags are: Enabled and Online Capable - in the FDT case we're always either enabled or disabled, but in ACPI we can be not-yet-enabled but hot-pluggable. The other flags, which I'd like to work in once this workstream starts maturing, are: Performance Interrupt Mode and VGIC Maintenance Interrupt Mode - these indicate that the corresponding interrupts are edge triggered (when set).